### DISCUSSION TITLE
averaging components

### DISCUSSION THREAD

**Discussion by wienecke** [flatironinstitute/CaImAn#1251] [Q&A]
Hi 
I use CNMF.fit, then CNMF.refit, to extract components from 2-photon GCaMP7f movies. 
I skip deconvolution by setting p to 0, and I use patches because SNR varies across the FOV.
CaImAn often extracts multiple components within a region of interest, and after df/f normalization, their relative strengths are lost. 
How should I use CaImAn to estimate df/f within an arbitrary region of interest? 

Currently I do the following:
1. Use estimates.A to select a subset of components with at least X% of their energy within the region of interest (X% is arbitrary).
2. Average estimates.C for that subset of components (since estimates.C have been scaled in estimates.normalize_components).
3. Pass that average to estimates.detrend_df_f to compute df/f normalized fluorescence.
Is this the procedure you recommend? 

Thank you!
Carl

**Comment by EricThomson** [comment]
I'm not sure about your exact pipeline, but I would just use the filters we already have in place: we have an SNR filter in component evaluation already, so just set that to a reasonable value, and then calculate dff using `detrend_df_f()` and you should be fine.
